src/main/plugin/iso19139.ca.HNAP/schematron/schematron-rules-common.sch

```diff
@@ -57,6 +57,33 @@
     </xsl:for-each>
   </xsl:function>
 
+  <xsl:key name="groupKey"
+    match="*//gmd:transferOptions/gmd:MD_DigitalTransferOptions/gmd:onLine/gmd:CI_OnlineResource[gmd:linkage/gmd:URL]"
+    use="concat(gmd:linkage/gmd:URL/text(), '|', gmd:description/gco:CharacterString/text())"/>
+
+  <xsl:function xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    name="geonet:getDuplicateResources"
+    as="xs:string">
+    <xsl:param name="elements" as="element()*"/>
+
+    <xsl:variable name="duplicateResources">
+      <xsl:for-each-group select="$elements"
+        group-by="concat(gmd:linkage/gmd:URL/text(), '|', gmd:description/gco:CharacterString/text())">
```

Comment:
`group-by="concat(gmd:linkage/gmd:URL/text(), '|', gmd:description/gco:CharacterString/text())">`
Should be
`group-by="concat(gmd:linkage/gmd:URL/text(), '|', tokenize(current-group()[1]/gmd:description/gco:CharacterString/text(), ';')[3])">`

Otherwise it will group based on the full description but the group by should be on the language.

Reply:
I have fix all related review and here is the final result looks like:

![image](https://github.com/metadata101/iso19139.ca.HNAP/assets/74916635/48e07e55-1e7f-499b-b0f1-61a1887bf660)